PMMain/Midas/MidasGUI/MidasThread.cpp

```diff
@@ -41,3 +42,97 @@ void MidasThread::threadEmitStateString(std::string str)
 {
     emit emitStateString(str);
 }
+
+
+std::vector<ringData> MidasThread::readKeyboardSetupFile()
+{
+    
+    std::ifstream readRingData("C:/Users/Owen/Documents/keyboardData.txt");
```

Comment:
We shouldn't hardcode the file path in this method. It should probably be passed as a parameter. Also, we shouldn't commit changes (or, at least, try and push changes) that have a specific absolute path. Always deal with relative paths.